libbeat/processors/add_cloud_metadata/provider_azure_vm.go

```diff
@@ -51,10 +85,147 @@ var azureVMMetadataFetcher = provider{
+		hfetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azHttpSchema, azMetadataURI)
+		if err != nil {
+			return hfetcher, fmt.Errorf("failed to create new http metadata fetcher: %w", err)
+		}
+		// fetcher represents an azure metadata fetcher. The struct includes two type of fetchers.
+		// 1. An http fetcher(hfetcher) which retrieves metadata from azure metadata endpoint and
+		// 2. A generic fetcher(gfetcher) which uses azure sdk to retrieve metadata of azure managed clusters.
+		fetcher, err := newAzureMetadataFetcher("azure", hfetcher)
+		if err != nil {
+			return fetcher, fmt.Errorf("failed to create new azure metadata fetcher: %w", err)
+		}
+		// gfetcher is created and assinged to fetcher after the fetcher is created in order the
+		// fetchAzureClusterMeta to be a method of fetcher. This is needed so that the generic fetcher
+		// can use the results/metadata that are already retrieved from http fetcher. SubscriptionId and
+		// resourceGroupName are then used to filter azure managed clusters results.
+		gfetcher, err := newGenericMetadataFetcher(config, "azure", azGenSchema, fetcher.fetchAzureClusterMeta)
+		if err != nil {
+			return fetcher, fmt.Errorf("failed to create new generic metadata fetcher: %w", err)
+		}
+		fetcher.genericMetadataFetcher = gfetcher
+		return fetcher, nil
 	},
 }
+
+// fetchMetadata fetches azure vm metadata from
+// 1. Azure metadata endpoint with httpMetadataFetcher
+// 2. Azure Managed Clusters using azure sdk  with genericMetadataFetcher
+func (az *azureMetadataFetcher) fetchMetadata(ctx context.Context, client http.Client) result {
+	res := result{provider: az.provider, metadata: mapstr.M{}, err: nil}
+	logger := logp.NewLogger("add_cloud_metadata")
+	httpRes := az.httpMetadataFetcher.fetchMetadata(ctx, client)
+	if httpRes.err != nil {
+		res.err = httpRes.err
+		return res
+	}
+	res.metadata = httpRes.metadata
+	az.httpMeta = httpRes.metadata
+	gRes := az.genericMetadataFetcher.fetchMetadata(ctx, client)
+	if gRes.err != nil {
+		logger.Warnf("Failed to get additional AKS Cluster meta: %+v", gRes.err)
+		return res
+	}
+
+	res.metadata.DeepUpdate(gRes.metadata)
+	return res
+}
+
+// getAzureCredentials returns credentials to connect to Azure
+// env vars TENANT_ID, CLIENT_ID and CLIENT_SECRET are required
+// if not set, NewDefaultAzureCredential method will be used
+func getAzureCredentials(logger *logp.Logger) (azcore.TokenCredential, error) {
+	if os.Getenv("TENANT_ID") != "" && os.Getenv("CLIENT_ID") != "" && os.Getenv("CLIENT_SECRET") != "" {
+		return azidentity.NewClientSecretCredential(os.Getenv("TENANT_ID"), os.Getenv("CLIENT_ID"), os.Getenv("CLIENT_SECRET"), nil)
+	} else {
+		logger.Debugf("No Client or Tenant configuration provided. Retrieving default Azure credentials")
+		return azidentity.NewDefaultAzureCredential(nil)
+	}
+}
+
+// getAKSClusterNameId returns the AKS cluster name and Id for a given resourceGroup
```

Comment:
```suggestion
// getAKSClusterNameId returns the AKS cluster name and ID for a given resourceGroup
```

Can you please do the same change everywhere (like in error messages). Either `ID`, `id`, or `identifier` are good.

Reply:
In the code as well, I recommend to change `Id` to `ID`. It is the Go convention. See https://go.dev/wiki/CodeReviewComments#initialisms.